maildir/maildir.c

```diff
@@ -549,7 +549,7 @@ static int maildir_mbox_check_stats(struct Mailbox *m, int flags)
   if (check_new || check_stats)
     maildir_check_dir(m, "cur", check_new, check_stats);
 
-  return 0;
+  return (m->msg_new > 0);
```

Comment:
`nitpick`: It'd be best to write this out to explicitly return an integer instead of a boolean. While I know a boolean value maps to an integer, the mixed return types can cause some confusion (at least in my case.) In addition, it'd be more consistent with other backends' implementations.

Reply:
Hmm... yeah.  I'm not quite sure what I was thinking.
I've changed `check_stats()` and wrappers and implementations to return `int` - number of new emails.